File: 백준/Silver/5426. 비밀 편지/비밀 편지.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int tc = Integer.parseInt(br.readLine());
        StringBuilder ans = new StringBuilder();
        while (tc-- > 0) {
            String s = br.readLine();

            int n = (int) Math.sqrt(s.length());

            for (int j = n-1; j >= 0; j--) {
                for (int i = 0; i < n; i++)
                    ans.append(s.charAt(i * n + j));
            }
            ans.append('\n');
        }

        System.out.println(ans);
    }
}

File: 백준/Silver/5426. 비밀 편지/비밀 편지.py
import sys
from math import sqrt
input = sys.stdin.readline

for _ in range(int(input())):
    s = input().rstrip()

    n = int(sqrt(len(s)))
    board = [[s[i*n+j] for j in range(n)] for i in range(n)]

    ans = []
    for j in range(n-1, -1, -1):
        for i in range(n):
            ans.append(board[i][j])

    print(''.join(ans))